crates/adapters/src/server/mod.rs

```diff
@@ -346,45 +383,190 @@ async fn shutdown(state: WebData<ServerState>) -> impl Responder {
                 }
                 HttpResponse::Ok().json("Pipeline terminated")
             }
-            Err(e) => HttpResponse::InternalServerError().json(&ErrorResponse::new(&format!(
-                "Failed to terminate the pipeline: {e}"
-            ))),
+            Err(e) => http_error(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                &format!("Failed to terminate the pipeline: '{e}'"),
+            ),
         }
     } else {
         HttpResponse::Ok().json("Pipeline already terminated")
     }
 }
 
-#[get("/input_endpoint/{endpoint_name}")]
-async fn input_endpoint(req: HttpRequest, stream: web::Payload) -> impl Responder {
-    match req.match_info().get("endpoint_name") {
-        None => HttpResponse::BadRequest().body("Missing endpoint name argument"),
-        Some(endpoint_name) => {
-            HttpInputTransport::get_endpoint_websocket(endpoint_name, &req, stream).unwrap_or_else(
-                |e| {
-                    HttpResponse::InternalServerError().json(&ErrorResponse::new(&format!(
-                        "Failed to establish connection to input HTTP endpoint: {e}"
-                    )))
-                },
-            )
+#[derive(Debug, Deserialize)]
+struct IngressArgs {
+    // #[serde(default = "HttpInputTransport::default_mode")]
+    // mode: HttpIngressMode,
+    #[serde(default = "HttpInputTransport::default_format")]
+    format: String,
+}
+
+#[post("/ingress/{table_name}")]
+async fn input_endpoint(
+    state: WebData<ServerState>,
+    req: HttpRequest,
+    args: Query<IngressArgs>,
+    payload: Payload,
+) -> impl Responder {
+    debug!("{req:?}");
+    let table_name = match req.match_info().get("table_name") {
+        None => return http_error(StatusCode::BAD_REQUEST, "Missing table name argument."),
+        Some(table_name) => table_name.to_string(),
+    };
+    // debug!("Table name {table_name:?}");
+
+    // Generate endpoint name.
+    let endpoint_name = format!("api-ingress-{table_name}-{}", Uuid::new_v4());
```

Comment:
We should probably use the same UUID types throughout. The pipeline manager uses `Uuid:now_v7()` for API object IDs.

Reply:
v7 unfortunately requires adding a `uuid_unstable` flag to `.cargo/config`, which is ok for the main repo, but is kind of painful for the auto-generated Rust workspace.

Reply:
Ack. Perhaps leave a comment noting this. We can flip it over to v7 when it's no longer unstable. I think @gz mentioned it should be stabilizing soon.

Reply:
I think `v7` is stable now, behind the `v7` feature flag. We may need to update `uuid` but it was stabilized in 1.3.4

Reply:
Released 2 days ago, nice! I'll fix this before merging the PR.

Reply:
Actually, I think it's still unstable in `1.3.4`.